eZ/Publish/Core/Persistence/Cache/AbstractInMemoryHandler.php

```diff
@@ -43,31 +42,43 @@ abstract class AbstractInMemoryHandler
     /**
      * Setups current handler with everything needed.
      *
-     * @param \Symfony\Component\Cache\Adapter\TagAwareAdapterInterface $cache
      * @param \eZ\Publish\SPI\Persistence\Handler $persistenceHandler
+     * @param \eZ\Publish\Core\Persistence\Cache\Adapter\InMemoryClearingProxyAdapter $cache
      * @param \eZ\Publish\Core\Persistence\Cache\PersistenceLogger $logger
      * @param \eZ\Publish\Core\Persistence\Cache\InMemory\InMemoryCache $inMemory
      */
     public function __construct(
-        TagAwareAdapterInterface $cache,
         PersistenceHandler $persistenceHandler,
+        InMemoryClearingProxyAdapter $cache,
         PersistenceLogger $logger,
         InMemoryCache $inMemory
     ) {
-        $this->cache = $cache;
         $this->persistenceHandler = $persistenceHandler;
-        $this->logger = $logger;
-        $this->inMemory = $inMemory;
-
-        $this->init();
+        $this->setCacheDependencies($cache, $logger, $inMemory);
     }
 
     /**
-     * Optional dunction to initialize handler without having to overload __construct().
+     * @api May be used from handlers that needs to inject something else then PersistenceHandler to __construct(), to
+     *      avoid having to have intimate knowledge of internal classes used by AbstractInMemoryHandler in signature.
```

Comment:
Hiding dependencies using argument unpacking not a good approach. If `AbstractInMemoryHandler` requires any internal dependency in the constructor then it should be internal as well.

Reply:
@adamwojs So should we?:
- keep init()
- mark constructor as internal
- anything else?

Reply:
> should be internal as well.

or what did you means here actually?